web/modules/custom/server_group/src/Plugin/EntityViewBuilder/NodeGroup.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+namespace Drupal\server_group\Plugin\EntityViewBuilder;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityRepositoryInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Routing\RedirectDestinationTrait;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Drupal\og\Og;
+use Drupal\og\OgAccessInterface;
+use Drupal\og\OgMembershipInterface;
+use Drupal\server_general\EntityViewBuilder\NodeViewBuilderAbstract;
+use Drupal\server_general\LineSeparatorTrait;
+use Drupal\server_general\SocialShareTrait;
+use Drupal\server_general\TitleAndLabelsTrait;
+use Drupal\user\Entity\User;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * The "Node Group" plugin.
+ *
+ * @EntityViewBuilder(
+ *   id = "node.group",
+ *   label = @Translation("Node - Group"),
+ *   description = "Node view builder for Group bundle."
+ * )
+ */
+class NodeGroup extends NodeViewBuilderAbstract {
+
+  use RedirectDestinationTrait;
+  use LineSeparatorTrait;
+  use SocialShareTrait;
+  use TitleAndLabelsTrait;
+
+  /**
+   * Constructor for NodeGroup.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entity_repository
+   *   The entity repository.
+   * @param \Drupal\og\OgAccessInterface $og_access
+   *   The og access.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, AccountInterface $current_user, EntityRepositoryInterface $entity_repository, OgAccessInterface $og_access) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $current_user, $entity_repository);
+    $this->ogAccess = $og_access;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('current_user'),
+      $container->get('entity.repository'),
```

Comment:
We already have those on the parent, so let's call the parent first and re-use it

Reply:
done